[CI] RemoteClusterConnectionTests.testDiscoverSingleNode

Doesn't reproduce locally

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+matrix-java-periodic/ES_BUILD_JAVA=java10,ES_RUNTIME_JAVA=java10,nodes=virtual&&linux/60/console

```
 2> REPRODUCE WITH: ./gradlew :server:test -Dtests.seed=EC9E0C70237EA138 -Dtests.class=org.elasticsearch.transport.RemoteClusterConnectionTests -Dtests.method="testDiscoverSingleNode" -Dtests.security.manager=true -Dtests.locale=ha -Dtests.timezone=America/Lower_Princes
ERROR   0.05s J0 | RemoteClusterConnectionTests.testDiscoverSingleNode <<< FAILURES!
   > Throwable #1: com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=1868, name=elasticsearch[org.elasticsearch.transport.RemoteClusterConnectionTests][management][T#1], state=RUNNABLE, group=TGRP-RemoteClusterConnectionTests]
   > 	at __randomizedtesting.SeedInfo.seed([EC9E0C70237EA138:1A50AFD86729C6D3]:0)
   > Caused by: java.lang.AssertionError: ExecutionCancelledException[operation was cancelled reason [connect handler is closed]]
   > 	at __randomizedtesting.SeedInfo.seed([EC9E0C70237EA138]:0)
   > 	at org.elasticsearch.transport.RemoteClusterConnectionTests$9.onFailure(RemoteClusterConnectionTests.java:963)
   > 	at org.elasticsearch.action.LatchedActionListener.onFailure(LatchedActionListener.java:50)
   > 	at org.elasticsearch.action.ActionListener.onFailure(ActionListener.java:138)
   > 	at org.elasticsearch.transport.RemoteClusterConnection$ConnectHandler$1.lambda$doRun$1(RemoteClusterConnection.java:407)
   > 	at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:68)
   > 	at org.elasticsearch.transport.RemoteClusterConnection$ConnectHandler.collectRemoteNodes(RemoteClusterConnection.java:474)
   > 	at org.elasticsearch.transport.RemoteClusterConnection$ConnectHandler$1.doRun(RemoteClusterConnection.java:412)
   > 	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
   > 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514)
   > 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
   > 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:625)
   > 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135)
   > 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
   > 	at java.base/java.lang.Thread.run(Thread.java:844)
   > Caused by: ExecutionCancelledException[operation was cancelled reason [connect handler is closed]]
   > 	at org.elasticsearch.common.util.CancellableThreads.onCancel(CancellableThreads.java:63)
   > 	at org.elasticsearch.common.util.CancellableThreads.executeIO(CancellableThreads.java:129)
   > 	at org.elasticsearch.transport.RemoteClusterConnection$ConnectHandler.collectRemoteNodes(RemoteClusterConnection.java:425)
   > 	... 8 more

```

looks similar to issue https://github.com/elastic/elasticsearch/issues/28695

@elastic/es-core-infra could please somebody have a look into it ?

--- a/server/src/test/java/org/elasticsearch/transport/RemoteClusterConnectionTests.java
+++ b/server/src/test/java/org/elasticsearch/transport/RemoteClusterConnectionTests.java
@@ -389,10 +389,27 @@ public class RemoteClusterConnectionTests extends ESTestCase {
         throws Exception {
         CountDownLatch latch = new CountDownLatch(1);
         AtomicReference<Exception> exceptionAtomicReference = new AtomicReference<>();
-        ActionListener<Void> listener = ActionListener.wrap(x -> latch.countDown(), x -> {
-            exceptionAtomicReference.set(x);
-            latch.countDown();
-        });
+        ActionListener<Void> listener = ActionListener.wrap(
+            x -> latch.countDown(),
+            x -> {
+                /*
+                 * This can occur on a thread submitted to the thread pool while we are closing the
+                 * remote cluster connection at the end of the test.
+                 */
+                if (x instanceof CancellableThreads.ExecutionCancelledException) {
+                    try {
+                        // we should already be shutting down
+                        assertEquals(0L, latch.getCount());
+                    } finally {
+                        // ensure we count down the latch on failure as well to not prevent failing tests from ending
+                        latch.countDown();
+                    }
+                    return;
+                }
+                exceptionAtomicReference.set(x);
+                latch.countDown();
+            }
+        );
         connection.updateSeedNodes(proxyAddress, seedNodes, listener);
         latch.await();
         if (exceptionAtomicReference.get() != null) {

